packages/app/src/server/routes/api/cortex/index.ts

```diff
@@ -31,6 +32,55 @@ export const cortexProxy = httpProxy.createProxyServer({ ignorePath: true });
  */
 cortexProxy.on("proxyReq", onProxyReq);
 
+cortexProxy.on("proxyRes", (proxyRes, req, res) => {
+  // if it is an HTML response, it is most likely an error message
+  const isHTMLErrorResponse = proxyRes.headers["content-type"]?.includes(
+    "text/html"
+  );
+  if (isHTMLErrorResponse) {
+  } else {
+    res.writeHead(proxyRes.statusCode!, proxyRes.headers);
+  }
+
+  const chunks: Array<any> = [];
+  proxyRes.on("data", function (chunk) {
+    if (isHTMLErrorResponse) {
+      /* In case of HTML error, we do not want to stream the response directly
+       * to the client, but instead extract the error message in `proxyRes.on("end")`
+       * and send that.
+       */
+      chunks.push(chunk);
+    } else {
+      res.write(chunk);
+    }
+  });
+  proxyRes.on("end", function () {
+    if (isHTMLErrorResponse) {
+      /* Error messages are returned via HTML via a specific layout.
+       * See examples here:
+       * - https://github.com/cortexproject/cortex/blob/82b32ec65ed16920e6053ac7fb748c42e3cae452/pkg/storegateway/gateway_http.go#L16-L25
+       * - https://github.com/cortexproject/cortex/blob/82b32ec65ed16920e6053ac7fb748c42e3cae452/pkg/compactor/compactor_http.go#L16-L25
+       */
+      const body = chunks.join();
+      const parsedHTML = parse(body);
+      const errorMessage = parsedHTML.querySelector("html body p")?.text;
```

Comment:
thanks for the changes : ) when code reads like an English sentence that says what it's doing w/o having to think a lot: :rocket: